src/logging.cpp

```diff
@@ -337,3 +374,26 @@ void BCLog::Logger::ShrinkDebugFile()
     else if (file != nullptr)
         fclose(file);
 }
+
+void BCLog::LogRateLimiter::MaybeReset()
+{
+    const std::chrono::seconds now = GetTime<std::chrono::seconds>();
+    if ((now - m_last_reset) >= WINDOW_SIZE) {
+        m_available_bytes = WINDOW_MAX_BYTES;
+        m_last_reset = now;
+        m_dropped_bytes = 0;
+    }
+}
+
+bool BCLog::LogRateLimiter::Consume(uint64_t bytes)
```

Comment:
Is it possible for this to return false followed by true, even if the rate limiter doesn't reset, eg:

- m_available_bytes is 100
- `Consume(110)` is called and returns false
- `Consume(90)` is called and returns true.

Which would then cause the "Restarting logging from ..." and "Excessive logging detected from..." logs to be called before the LogRateLimiter has been reset.

If so, I think this is the fix:

```diff
    if (bytes > m_available_bytes) {
        m_dropped_bytes += bytes;
+       m_available_bytes = 0;
        return false;
    }

Reply:
Yes, trying to consume more bytes than are available should result in no available bytes afterwards.